pennylane/devices/default_qubit_tf.py

```diff
@@ -145,10 +143,19 @@ class DefaultQubitTF(DefaultQubit):
     _roll = staticmethod(tf.roll)
     _stack = staticmethod(tf.stack)
 
+    @staticmethod
+    def _const_mul(constant, array):
+        return constant * array
+
     @staticmethod
     def _asarray(array, dtype=None):
+        if isinstance(array, tf.Tensor):
+            if dtype is None or dtype == array.dtype:
+                return array
+            return tf.cast(array, dtype)
```

Comment:
For my understanding: we do this here so that `tf.convert_to_tensor` doesn't need to be called for sure?

Reply:
It was due to the cases that `tf.convert_to_tensor` raises an error. See e.g. https://stackoverflow.com/questions/35596629/how-to-convert-tf-int64-to-tf-float32